internal/resources/data_source_instance_storage_controller.go

```diff
@@ -32,7 +32,7 @@ func ReadInstanceStorageController() *schema.Resource {
 			"interface_number": {
 				Type:        schema.TypeInt,
 				Required:    true,
-				Description: "The interface number to be allocated",
+				Description: "The interface number to be allocated. Enter `0` to automatically pick the next available interface number.",
```

Comment:
not a good user experience if we say specify zero. Can it be not passed if its  optional ?